anvil/server/src/ipc.rs

```diff
@@ -0,0 +1,178 @@
+//! IPC handling
+
+use crate::{error::RequestError, pubsub::PubSubConnection, PubSubRpcHandler};
+use anvil_rpc::request::Request;
+use bytes::BytesMut;
+use futures::{ready, Sink, Stream, StreamExt};
+use parity_tokio_ipc::Endpoint;
+use std::{
+    future::Future,
+    io,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tracing::{error, trace, warn};
+
+/// An IPC connection for anvil
+///
+/// A Future that listens for incoming connections and spawns new connections
+pub struct IpcEndpoint<Handler> {
+    /// the handler for the websocket connection
+    handler: Handler,
+    /// The endpoint we listen for incoming transactions
+    endpoint: Endpoint,
+    // TODO add shutdown
```

Comment:
ah forgot about this,

not required, since does not to do any shutdown handling since everything is converted into a task that is managed by the `NodeHandle`